src/cbdgen-framework.py

```diff
@@ -132,58 +184,84 @@ def main(seed=None):
     logbook = tools.Logbook()
     logbook.header = "gen", "evals", "std", "min", "avg", "max"
 
-    pop = toolbox.population(POP)
+    tool_pop = toolbox.population(pop)
 
     # Evaluate the individuals with an invalid fitness
-    invalid_ind = [ind for ind in pop if not ind.fitness.valid]
+    invalid_ind = [ind for ind in tool_pop if not ind.fitness.valid]
     fitnesses = toolbox.map(toolbox.evaluate, invalid_ind)
     for ind, fit in zip(invalid_ind, fitnesses):
         ind.fitness.values = fit
     # Compile statistics about the population
-    record = stats.compile(pop)
+    record = stats.compile(tool_pop)
 
     logbook.record(gen=0, evals=len(invalid_ind), **record)
     print(logbook.stream)
     # Begin the generational process
-    for gen in range(1, NGEN):
-        offspring = algorithms.varAnd(pop, toolbox, CXPB, MUTPB)
+    for gen in range(1, ngen):
+        offspring = algorithms.varAnd(tool_pop, toolbox, cxpb, mutpb)
         # Evaluate the individuals with an invalid fitness
         invalid_ind = [ind for ind in offspring if not ind.fitness.valid]
         fitnesses = toolbox.map(toolbox.evaluate, invalid_ind)
         for ind, fit in zip(invalid_ind, fitnesses):
             ind.fitness.values = fit
         # Select the next generation population from parents and offspring
-        pop = toolbox.select(pop + offspring, POP)
+        tool_pop = toolbox.select(tool_pop + offspring, pop)
 
         # Compile statistics about the new population
-        record = stats.compile(pop)
+        record = stats.compile(tool_pop)
         logbook.record(gen=gen, evals=len(invalid_ind), **record)
         print(logbook.stream)
-    return pop, logbook
+    return tool_pop, logbook
+
+def main():
+    options = setup.get_options()
+
+    if options['filepath'] != '':
+        base_df = pd.read_csv(options['filepath'])
+
+    global dataFrame
```

Comment:
Global keywords or global variables should be minimized when working in a modular architecture software. Varibles `dataFrame`, `metrics` and `global_measures` should be improved to minimize the use of global variables.

Reply:
Addressed in #36